Domains/Jobs/WorkerConfiguration.php

```diff
@@ -0,0 +1,338 @@
+     *
+     * @var int
+     */
+    protected $timeout = 30;
+
+    /**
+     * Job's worker sleep configuration, as the number of seconds to wait/sleep in between polling the queue.
+     *
+     * @var int
+     */
+    protected $sleep = 3;
+
+    /**
+     * Number of times to attempt a job before logging it failed
+     *
+     * @var int
+     */
+    protected $tries = 3;
+
+    /**
+     * Job's worker force configuration, indicates if the worker should run even in Laravel's maintenance mode.
+     *
+     * @var bool
+     */
+    protected $force = false;
+
+    /**
+     * Job's worker stopWhenEmpty configuration, indicates if the worker should stop when queue is empty.
+     *
+     * @var bool
+     */
+    protected $stopWhenEmpty = false;
+
+    /**
+     * The number of jobs to process before stopping
+     *
+     * @var int
+     */
+    protected $maxJobs = 0;
+
+
+    /**
+     * The maximum number of seconds the worker should run
+     *
+     * @var int
+     */
+    protected $maxTime = 0;
+
+
+    /**
+     * Number of seconds to rest between jobs
+     *
+     * @var int
+     */
+    protected $rest = 0;
+
+    /**
+     * Set the configurations
+     *
+     * @param array $options
+     * @return self
+     */
+    public static function withOptions(array $options = []) : self
+    {
+        $self = new static;
+
+        collect($options)
+            ->each(fn($value, $option) => method_exists($self, 'set' . ucfirst($option)) 
+                ? $self->{'set' . ucfirst($option)}($value) 
+                : null
+            );
+        
+        return $self;
+    }
+
+    /**
+     * Get the worker options
+     *
+     * @return array
+     */
+    public function getWorkerOptions() : array 
+    {
+        return [
+            $this->getName(),
+            $this->getBackoff(),
+            $this->getMemory(),
+            $this->getTimeout(),
+            $this->getSleep(),
+            $this->getTries(),
+            $this->getForce(),
+            $this->getStopWhenEmpty(),
+            $this->getMaxJobs(),
+            $this->getMaxTime(),
+            $this->getRest(),
+        ];
+    }
+
+    /**
+     * Set the worker name
+     */
+    public function setName(string $name): self
+    {
+        $this->name = $name;
+
+        return $this;
+    }
+
+    /**
+     * Get the worker name
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+    
+    /**
+     * The number of seconds to wait before retrying the job
+     */
+    public function setBackoff(int $value): self
+    {
+        $this->backoff = $value;
+
+        return $this;
+    }
+
+    /**
+     * Get the number of seconds to wait before retrying the job
+     */
+    public function getBackoff(): int
+    {
+        return $this->backoff;
+    }
+
+    /**
+     * The maximum amount of RAM the worker may consume.
```

Comment:
Maybe specify in the self-documentation: is the unit in MB?

Reply:
Done.